fedimint-core/src/amount.rs

```diff
@@ -0,0 +1,295 @@
+use std::num::ParseIntError;
+use std::str::FromStr;
+
+use anyhow::bail;
+use bitcoin::Denomination;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::encoding::{Decodable, Encodable};
+
+pub const SATS_PER_BITCOIN: u64 = 100_000_000;
+
+/// Shorthand for [`Amount::from_msats`]
+pub fn msats(msats: u64) -> Amount {
+    Amount::from_msats(msats)
+}
+
+/// Shorthand for [`Amount::from_sats`]
+pub fn sats(amount: u64) -> Amount {
+    Amount::from_sats(amount)
+}
+
+/// Represents an amount of BTC. The base denomination is millisatoshis, which
+/// is why the `Amount` type from rust-bitcoin isn't used instead.
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Hash,
+    Deserialize,
+    Serialize,
+    Encodable,
+    Decodable,
+)]
+#[serde(transparent)]
+pub struct Amount {
+    pub msats: u64,
+}
```

Comment:
food for thought: maybe in the future we could rely on [rust-bitcoin Amount](https://docs.rs/bitcoin/latest/bitcoin/struct.Amount.html) instead 🤔

Reply:
oh, I should've read the docstring first lol, forgot about the msats issue